components/public-api/gitpod/v1/workspace.proto

```diff
@@ -157,6 +175,70 @@ message StartWorkspaceResponse {
   Workspace workspace = 1;
 }
 
+message StopWorkspaceRequest {
+  // workspace_id specifies which workspace should be stopped.
+  //
+  // +required
+  string workspace_id = 1;
+}
+
+message StopWorkspaceResponse {}
+
+message DeleteWorkspaceRequest {
+  // workspace_id specifies the workspace that is going to delete.
+  //
+  // +required
+  string workspace_id = 1;
+}
+
+message DeleteWorkspaceResponse {}
+
+message UpdateWorkspaceRequest {
+  // workspace_id specifies the workspace to update.
+  //
+  // +required
+  string workspace_id = 1;
+
+  optional string name = 2;
+
+  // Admission describes who can access a workspace instance and its ports.
+  optional AdmissionLevel admission = 3;
+
+  optional bool pinned = 4;
+}
+
+message UpdateWorkspaceResponse {
+  Workspace workspace = 1;
+}
+
+message UpdateWorkspaceStatusRequest {
+  // workspace_id specifies the workspace to update.
+  //
+  // +required
+  string workspace_id = 1;
+
+  // timeout is the timeout duration to stop the workspace since the last
+  // activity.
+  optional google.protobuf.Duration timeout = 5;
```

Comment:
I'm not sure it should be here, it is desired state, not status. It is probably better to have `Set/GetWorkspaceTimeout`

cc @csweichel @svenefftinge

Reply:
how do we get workspace timeout right now btw?

Reply:
We don't get workspace timeout in dashboard, so no need to address it now for migration actually

Reply:
I'm mostly trying to undertand how our API should work conceptually and be complete

Reply:
UpdateWorkspaceStatus shoudl be called by ws-manager/workspace to give an actual status, not a user to provide specification. I think we should Have `UpdateWorkspaceTimeout` and `GetWorkspaceTimeout` for that.

We should add JS-doc on `UpdateWorkspaceStatus` that it should no be called by a user.